NetworkTraffic: Simplify and Cleanup (Revised)

gj86 opened the pull request:
* Improve code readability.

* Put getting total Rx/Tx data in a function.

* Wrap checking for aboveThreshold in a function,
  also convert speed to Kilo/Speed before check.

* Simplify handleMessage() & setTrafficDrawable().
  Since we are not showing 2 way traffic together
  anymore, there is no point for this if/else checks.

* Remove an extra # from 100 * Mega check, we will
  show speed in Giga if speed is >999 Mega anyway.

* Modify "0.00" pattern to "0.##" for Kilo speed scale.
  This will show "0 KB/s" instead of "0.00 KB/s".

  Decimal separator does not show if there is no
  fraction in formated number.

Signed-off-by: Gobinda Joy <gobinda.joy@gmail.com>

neobuddy89 commented on packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java:
```diff
@@ -290,7 +273,7 @@ private CharSequence formatOutput(long speed) {
                 formatSpeed =  decimalFormat.format(speed / (float)Kilo);
             } else {
-                decimalFormat = new DecimalFormat("0.00");
+                decimalFormat = new DecimalFormat("0.##");
```
THis will show DOT after number 5. kbps instead 5 kbps.

gj86 replied:
Nope. Dot is removed from formated string if there is no fraction. Java DecimalFormat does this automatically.
Try this here [jdoodle](https://www.jdoodle.com/online-java-compiler/)

> import java.text.DecimalFormat;
> 
> public class MyClass {
>     public static void main(String args[]) {
>       DecimalFormat df = new DecimalFormat("0.##");
>       System.out.println(df.format(5));
>     }
> }

neobuddy89 commented on packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java:
```diff
@@ -270,7 +253,7 @@ private CharSequence formatOutput(long speed) {
                 formatSpeed =  decimalFormat.format(speed / (float)Giga);
             } else if (speed >= 100 * Mega) {
-                decimalFormat = new DecimalFormat("###0");
+                decimalFormat = new DecimalFormat("##0");
```
why is # reduced here?

gj86 replied:
\# is not needed here as 1000 Mega/1 Giga will match the if (Giga).

gj86 commented on packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java:
```diff
@@ -69,20 +69,22 @@
+    protected int mLocation = LOCATION_DISABLED;
     private int mMode = MODE_UPSTREAM_AND_DOWNSTREAM;
     private int mSubMode = MODE_UPSTREAM_AND_DOWNSTREAM;
     private boolean mConnectionAvailable;
```
No need for this anymore.

gj86 commented on packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java:
```diff
@@ -155,80 +157,61 @@ protected void onDetachedFromWindow() {
+            final boolean aboveThreshold = isAboveThreshold(rxBytes, txBytes);
+            int submode = MODE_DOWNSTREAM_ONLY;
+
             mIsActive = enabled && mAttached && (!mAutoHide || (mConnectionAvailable && aboveThreshold));
```
Need to remove `mConnectionAvailable`

gj86 committed:
NetworkTraffic: Simplify and Cleanup (Revised)

* Improve code readability.

* Put getting total Rx/Tx data in a function.

* Wrap checking for aboveThreshold in a function,
  also convert speed to Kilo/Speed before check.

* Simplify handleMessage() & setTrafficDrawable().
  Since we are not showing 2 way traffic together
  anymore, there is no point for this if/else checks.

* Remove an extra # from 100 * Mega check, we will
  show speed in Giga if speed is >999 Mega anyway.

* Modify "0.00" pattern to "0.##" for Kilo speed scale.
  This will show "0 KB/s" instead of "0.00 KB/s".

  Decimal separator does not show if there is no
  fraction in formated number.

Signed-off-by: Gobinda Joy <gobinda.joy@gmail.com>
